src/main/java/org/folio/innreach/repository/InnReachTransactionRepository.java

```diff
@@ -24,8 +25,8 @@
   @Query(name = InnReachTransaction.FETCH_ONE_BY_ID_QUERY_NAME)
   Optional<InnReachTransaction> fetchOneById(UUID id);
 
-  @Query(name = InnReachTransaction.FETCH_ONE_BY_ITEM_BARCODE_QUERY_NAME)
-  Optional<InnReachTransaction> fetchOneByFolioItemBarcode(String itemBarcode);
+  @Query(name = InnReachTransaction.FETCH_ONE_BY_ITEM_BARCODE_AND_STATES_QUERY_NAME)
+  Optional<InnReachTransaction> fetchOneByFolioItemBarcodeAndStates(String itemBarcode, List<InnReachTransaction.TransactionState> states);
```

Comment:
I think `Set` for states would be more appropriate here instead of `List`
To populate parameters you can use EnumSet as a specialized implementation of Set for enum values